act-rules-format/act-rules-format.bs

```diff
@@ -656,6 +705,16 @@ Definitions {#definitions}
     <p>A document, such as a standard, contract, policy or regulation, that includes [=accessibility requirements=]. For example, WCAG 2.1, WAI-ARIA 1.1, HTML 5.2, EPUB Accessibility 1.0, BBC HTML Accessibility Standards v2.0</p>
   </dd>
 
+  <dt><dfn>Implementation</dfn><dt>
+  <dd>
+    <p>An accessibility test methodology or test tool, containing a set of [=implementation procedures=] that describes how to get the expected [=outcomes=] of an ACT Rule.</p>
+  </dd>
+
+  <dt><dfn>Implementation Procedure</dfn><dt>
+  <dd>
+    <p>A procedure resulting in an [=outcome=]. The procedure can be a step by step description of how to manually perform the test, a fully automated test, or some combination of manual and automated testing.</p>
+  </dd>
```

Comment:
(⚠️ set theory / computability vocabulary included)
So, a (mathematical or computer program) function can either be described as an _intention_, i.e. a formula or code to compute it (e.g. `x => x*x` is an intention for the squaring function); or it can be described as an _extension_, i.e. the set of (input, output) pairs (e.g. `{ (0, 0), (1, 1), (2, 4), ... }`for the squaring function.
Intentions are usually easier to manipulate and we are used to them. Extensions are sometimes useful to consider.

Anyway, here the procedure/check is described as an intention ("a step by step procedure"). But ultimately, the only thing that matters for consistency is the outcomes it gives, i.e. its extension.
So, I'm wondering whether it could make sense to instead define the procedure/check as "a set of outcomes" rather than as the piece of code that produces this set.
It may be a bit harder to grasp at first, but it might allow to simplify the other definitions down the line. It is also a bit closer to what we do in practice since we work with the implementation reports (an extension) rather than the code itself (an intension).